R/list-simplify.R

```diff
@@ -0,0 +1,40 @@
+list_simplify <- function(x, simplify = NA, ptype = NULL) {
+  vec_assert(x, list())
```

Comment:
I'd create our own assertion here because `vec_assert()` is still experimental and needs more work. For instance this will fail with S3 objects that explictly inherit from `"list"`.

I think we'd disallow data frames and check for `vec_is_list()`?